src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        WorkWithFile workWithFile = new WorkWithFile();
+        String report = workWithFile.createReportFromCsvFile(fromFileName);
+        workWithFile.writeToFile(toFileName, report);
+    }
+
+    private String createReportFromCsvFile(String fileName) {
+        int totalBuy = 0;
+        int totalSupply = 0;
+        final int buy = 0;
+        final int supply = 0;
+        final int amount = 1;
+        final String separator = ",";
+        final String supplyString = "supply";
+        final String buyString = "buy";
+        String line;
+        StringBuilder info = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] word = line.split(separator);
+                if (word[buy].equals(buyString)) {
+                    totalBuy += Integer.parseInt(word[amount]);
+                }
+                if (word[supply].equals(supplyString)) {
+                    totalSupply += Integer.parseInt(word[amount]);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file" + fileName, e);
+        }
+        info.append(supplyString).append(separator).append(totalSupply)
```

Comment:
It could be done by calling append chain on info only once, just: info.apend(...).append(..)... Not info.append(...); info.append(...);